src/Functions/array/arrayIntersect.cpp

```diff
@@ -344,8 +344,7 @@ FunctionArrayIntersect::UnpackedArrays FunctionArrayIntersect::prepareArrays(
                 if (isInteger(nested_init_type)
                     || isDate(nested_init_type)
                     || isDateTime(nested_init_type)
-                    || isDateTime64(nested_init_type)
-                    || isDecimal(nested_init_type))
```

Comment:
@alexey-milovidov This line led to the test failure in previous runs. It was added in [2b3c423](https://github.com/ClickHouse/ClickHouse/pull/47594/commits/2b3c42335a57533b4a06bbf57803dcbf202509ad). To understand whether it's a correct fix or not, I need a bit more digging. But probably you can tell it w/o one.

Reply:
Not sure, but the tests have passed. Thank you!